### DISCUSSION TITLE
Get Current Context - OpenTelemetry Java

### DISCUSSION THREAD

**Discussion by Akrammousselmal** [open-telemetry/opentelemetry-java#5313] [Q&A]
Hi,

I currently have a service that runs automated instrumentation configured using ENV variable in java. I am now receiving traceParent/Tracestate from another service through pub/sub messaging and would like to create a new span using this context to enable distributed tracing for some transactions. My issue here is that we never used manual instrumentation with OpenTelemetry in this legacy project. (We are using version 0.15.0)

How can I get the current context/OpenTelemetry object without creating it from scratch like explained here https://opentelemetry.io/docs/instrumentation/java/manual/

I would have expected the auto-instrumentation to have some sort of a singleton when switching to both auto/manual instrumentation.

**Comment by jack-berg** [comment]
The current context is accessible via `Context.current()`.

A global `OpenTelemetry` instance is configured when using the otel java agent, and is accessible in manual instrumentation using `GlobalOpenTelemetry.get()`.
